lib/apartment/adapters/abstract_adapter.rb

```diff
@@ -80,10 +85,12 @@ def drop(tenant)
       #   @param {String} tenant name
       #
       def switch!(tenant = nil)
-        return reset if tenant.nil?
+        run_callbacks :switch do
```

Comment:
Hm. `#switch` uses `#switch!` internally, so I think `#switch!` is sufficient, unless we want to distinguish callbacks for `#switch` vs `#switch!`. I'm not sure it'd make a difference to what I'm trying to do, though. Thoughts?

Note, though, a call to `#switch` will run the callbacks twice, once when it switches into the new tenant and once when it reverts back to the previous one. This behavior is useful for my purposes, but could be unexpected without knowing the internals of `#switch`.